### PR TITLE
[CBRD-23074] [log consumer refactoring] stream fetcher

### PR DESCRIPTION
**Pull request by radudoros**
http://jira.cubrid.org/browse/CBRD-23074

Comprising changes:
1. Refactor log_consumer_daemon logic outside of log_consumer.hpp/cpp into stream_entry_fetcher files.
2. Abstract m_stream_entries queue + mutex logic + into a concurrent_queue data structure.

### CODE REVIEW

File: src/replication/log_consumer.cpp [hunk 1]
```diff
@@ -310,82 +279,33 @@ namespace cubreplication
...
   void log_consumer::start_daemons (void)
   {
     m_subtran_applier = new subtran_applier (*this);
 
     er_log_debug_replication (ARG_FILE_LINE, "log_consumer::start_daemons\n");
-    m_consumer_daemon = cubthread::get_manager ()->create_daemon (cubthread::delta_time (0),
-			new consumer_daemon_task (*this),
-			"prepare_stream_entry_daemon");
+
+    m_entry_fetcher = new cubstream::stream_entry_fetcher ([this] (stream_entry * se)
```
Comment (razvanradulescu) [inline src/replication/log_consumer.cpp]: It may be more readable to define the lambda before.

File: src/replication/stream_entry_fetcher.cpp [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      stream_entry_fetch_task (stream_entry_fetcher &fetcher)
+	: m_fetcher (fetcher)
+      {
+      };
+
+      void execute (cubthread::entry &thread_ref) override
+      {
+	stream_entry *se = NULL;
+
+	int err = m_fetcher.fetch_stream_entry (se);
```
Comment (razvanradulescu) [inline src/replication/stream_entry_fetcher.cpp]: This code may be included in a single method of fetcher 
fetch_stream_entry + m_on_fetch  + push_entry = produce
The new metthod produce may be protected with stream_entry_fetch_task class fried of stream_entry_fetcher.
the other may be private

File: src/replication/stream_entry_fetcher.hpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#include "log_consumer.hpp"
+#include "replication_stream_entry.hpp"
+#include "concurrent_queue.hpp"
+
+namespace cubstream
+{
+  using cubreplication::log_consumer;
+  using cubreplication::stream_entry;
+
+  class stream_entry_fetcher
```
Comment (razvanradulescu) [inline src/replication/stream_entry_fetcher.hpp]: I was hoping to not have a dependency on replication_stream_entry.
If this dependency is kept, we need to change namespace as cubreplication.

If possible, make stream_entry_fetcher a template, and will be instantiated with replication_stream_entry on creation.
We might have issue with the m_on_fetch lambda.

File: src/replication/stream_entry_fetcher.hpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  {
+    public:
+      stream_entry_fetcher (const std::function<void (stream_entry *)> &on_fetch, cubstream::multi_thread_stream &stream);
+      ~stream_entry_fetcher ();
+
+      int fetch_stream_entry (stream_entry *&entry);
```
Comment (razvanradulescu) [inline src/replication/stream_entry_fetcher.hpp]: if possible, to make these private or protected (fetch_stream_entry  + push_entry )
Please see comment above related to stream_entry_fetch_task.

**Comment by radudoros** [comment]
I seem to have some regressions.. I'll need to check them

**Comment by radudoros** [comment]
Closed due to discussions about design (use . Started In progress pr https://github.com/CUBRID/cubrid/pull/1766

**Pull request closed without merging**